app/repository/sbrctrl/RestAdminDataUnitLinksEditRepository.scala

```diff
@@ -6,16 +6,46 @@ import com.typesafe.scalalogging.LazyLogging
 import repository.DataSourceNames._
 import repository._
 import repository.rest.Repository
-import uk.gov.ons.sbr.models.{ Period, VatRef }
+import uk.gov.ons.sbr.models.UnitType.{ LegalUnit, ValueAddedTax }
+import uk.gov.ons.sbr.models.{ Period, UnitId, UnitKey, VatRef }
 import uk.gov.ons.sbr.models.edit.Patch
-import unitref.VatUnitRef
 
 import scala.concurrent._
+import scala.concurrent.ExecutionContext.Implicits.global
+
+case class UrlAndPatch(url: String, patch: Patch)
+object UrlAndPatch {
+  def apply(urlAndPatch: (String, Patch)): UrlAndPatch = UrlAndPatch(urlAndPatch._1, urlAndPatch._2)
+}
 
 class RestAdminDataUnitLinksEditRepository @Inject() (@Named(SbrCtrl) unitRepository: Repository) extends AdminDataUnitLinksEditRepository with LazyLogging {
```

Comment:
This __repository__ seems to have a great deal of understanding about patches:
* it knows how to poke around inside a patch and extract values to form a key
* there is an implicit / hidden understanding  between different bits of code that the "first" patch is special ...

It does not feel as if any of this should be the responsibility of the repository - this should be handled at the service layer.  

At present the service is creating a patch - only for later code to have to parse that patch to understand what to do.  The service should have all the understanding - and instruct the repository what to do.  The repository should just be a dumb abstraction over the database / REST.

The service should be responsible for translating the input EditParent request into 3 separate repository calls.  It will co-ordinate these calls, and has the knowledge that the update should be performed first.  This avoids any implicit knowledge between objects that "the first thing in the list is special".

We should talk about this more - happy to work with you to come up with a design.
Note that I have not reviewed a lot of what follows in detail, as it is heavily tied to the current approach.